rust/crd/src/lib.rs

```diff
@@ -124,9 +212,18 @@ pub enum KafkaRole {
     Broker,
 }
 
-#[derive(Clone, Debug, Deserialize, Eq, JsonSchema, PartialEq, Serialize, Default)]
+#[derive(Clone, Debug, Default, Deserialize, Merge, JsonSchema, PartialEq, Serialize)]
 #[serde(rename_all = "camelCase")]
-pub struct KafkaConfig {}
+pub struct Storage {
+    #[serde(default)]
+    pub log_dirs: PvcConfig,
+}
+
+#[derive(Clone, Debug, Deserialize, JsonSchema, PartialEq, Serialize, Default)]
+#[serde(rename_all = "camelCase")]
+pub struct KafkaConfig {
+    pub resources: Option<Resources<Storage, NoRuntimeLimits>>,
```

Comment:
I think this can just be `#[serde(default)]` without the `Option<T>`

Reply:
Shouldn't this be `JvmRuntimeLimits` btw?

Reply:
I would have assumed so, but i kept to how it was implemented in HDFS and ZooKeeper to be consistent. Does it even safe resources to specifiy a min value?

Reply:
done